perception/traffic_light_classifier/src/single_image_debug_inference_node.cpp

```diff
@@ -0,0 +1,164 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <rclcpp/rclcpp.hpp>
+
+#if ENABLE_GPU
+#include <traffic_light_classifier/cnn_classifier.hpp>
+#endif
+
+#include <traffic_light_classifier/color_classifier.hpp>
+#include <traffic_light_classifier/nodelet.hpp>
+
+#include <memory>
+#include <string>
+
+namespace
+{
+std::string toString(const uint8_t state)
+{
+  if (state == autoware_auto_perception_msgs::msg::TrafficLight::RED) {
+    return "red";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::AMBER) {
+    return "yellow";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::GREEN) {
+    return "green";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::WHITE) {
+    return "white";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::CIRCLE) {
+    return "circle";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::LEFT_ARROW) {
+    return "left";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::RIGHT_ARROW) {
+    return "right";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::UP_ARROW) {
+    return "straight";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::DOWN_ARROW) {
+    return "down";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::DOWN_LEFT_ARROW) {
+    return "down_left";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::DOWN_RIGHT_ARROW) {
+    return "down_right";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::CROSS) {
+    return "cross";
+  } else if (state == autoware_auto_perception_msgs::msg::TrafficLight::UNKNOWN) {
+    return "unknown";
+  } else {
+    return "";
+  }
+}
+}  // namespace
+
+namespace traffic_light
+{
+class SingleImageDebugInferenceNode : public rclcpp::Node
+{
+public:
+  explicit SingleImageDebugInferenceNode(const rclcpp::NodeOptions & node_options)
+  : Node("single_image_debug_inference", node_options)
+  {
+    const auto image_path = declare_parameter("image_path", "");
+
+    int classifier_type = this->declare_parameter(
+      "classifier_type",
+      static_cast<int>(TrafficLightClassifierNodelet::ClassifierType::HSVFilter));
+    if (classifier_type == TrafficLightClassifierNodelet::ClassifierType::HSVFilter) {
+      classifier_ptr_ = std::make_unique<ColorClassifier>(this);
+    } else if (classifier_type == TrafficLightClassifierNodelet::ClassifierType::CNN) {
```

Comment:
I really love this debug tool but in order to use the CNN classifier, shouldn't we give the default label and onnx file as parameters?  these params don't declare in this node, and the default label file is just "label.txt".